### DISCUSSION TITLE
How to design a backend system

### DISCUSSION THREAD

**Discussion by ulut0002** [muxinc/next-video#296] [Q&A]
Hi,

Thank you for creating such a fantastic package—it’s been incredibly helpful!

I successfully hosted videos by importing them directly, like this: `import MasksOnATable from "@/videos/MasksOnATable.mp4";`. Everything worked seamlessly.

However, I’m now working on designing a backend system for my NextJS app that needs to manage a collection of video content. Each item in this collection will include a video source and a description, which I plan to use for filtering and sorting purposes.

Here’s the scenario: Suppose my client provides me with 100 videos. I want to upload these videos to the server, and I expect to see 100 JSON files and corresponding video files stored in the /videos directory. My challenge is figuring out how to reference these videos in my MongoDB collection. Specifically:

1. How should I structure my MongoDB documents to reference the video files effectively?
2. When I retrieve a result set from the server, how can I identify and link the correct video file in my code?

For example, if I have the following data structure:

```
data: [{
  type: "video",
  title: "Lorem Ipsum",
  src: "MasksOnATable"
}]
```

How can I ensure that the src property correctly identifies and links to the video file? What would be the best approach to implement this in a scalable way?

Thank you for your guidance!

**Comment by luwes** [comment]
Hi, we recently released v1.2.0 which provides hooks in the config thanks to a user contribution!
https://github.com/muxinc/next-video/pull/281

They get called when saving, updating and loading an asset and you could customize how and where the JSON is stored and loaded from. Or give it a unique ID.

Hope this helps. Let me know if you have any more questions.
